src/main/java/uk/gov/hmcts/reform/sscs/service/RoboticsJsonUploadService.java

```diff
@@ -0,0 +1,129 @@
+package uk.gov.hmcts.reform.sscs.service;
+
+import static java.util.Collections.singletonList;
+import static org.springframework.http.MediaType.*;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections.CollectionUtils;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.multipart.MultipartFile;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.document.DocumentUploadClientApi;
+import uk.gov.hmcts.reform.document.domain.UploadResponse;
+import uk.gov.hmcts.reform.sscs.ccd.domain.*;
+import uk.gov.hmcts.reform.sscs.ccd.exception.CcdException;
+import uk.gov.hmcts.reform.sscs.ccd.service.CcdService;
+import uk.gov.hmcts.reform.sscs.domain.pdf.ByteArrayMultipartFile;
+import uk.gov.hmcts.reform.sscs.exception.UnsupportedDocumentTypeException;
+import uk.gov.hmcts.reform.sscs.idam.IdamTokens;
+
+@Service
+@Slf4j
+public class RoboticsJsonUploadService {
+    private static final String DM_STORE_USER_ID = "sscs";
+    private static final String S2S_TOKEN = "oauth2Token";
+    private static final String ROBOTICS_JSON_FILENAME = "robotics_json.txt";
+
+    private final CcdService ccdService;
+    private final DocumentUploadClientApi documentUploadClientApi;
+    private final AuthTokenGenerator authTokenGenerator;
+
+    @Autowired
+    RoboticsJsonUploadService(
+            DocumentUploadClientApi documentUploadClientApi,
+            AuthTokenGenerator authTokenGenerator,
+            CcdService ccdService) {
+
+        this.documentUploadClientApi = documentUploadClientApi;
+        this.authTokenGenerator = authTokenGenerator;
+        this.ccdService = ccdService;
+    }
+
+    public void updateCaseWithRoboticsJson(
+            JSONObject roboticsJson,
+            SscsCaseData caseData,
+            SscsCaseDetails caseDetails,
+            IdamTokens idamTokens) {
+
+        ByteArrayMultipartFile file = ByteArrayMultipartFile.builder()
+                .content(roboticsJson.toString().getBytes())
+                .name(ROBOTICS_JSON_FILENAME)
+                .contentType(TEXT_PLAIN)
+                .build();
+
+        log.info("Uploading Robotics JSON for case {}", caseDetails.getId());
+        UploadResponse uploadResponse = uploadRoboticsJson(singletonList(file));
+
+        log.info("Attaching Robotics JSON to case {}", caseDetails.getId());
+        SscsCaseData updatedCaseData = attachRoboticsJsonToCaseData(caseData, uploadResponse);
+
+        try {
+            ccdService.updateCase(updatedCaseData, caseDetails.getId(),
+                    "attachRoboticsJson", "", "", idamTokens);
+        } catch (CcdException ccdEx) {
```

Comment:
here we have to catch any Exception rather than only the CcdException and then carry on as you currently do. And I wouldn't log it as an ERROR since that will trigger an alert. I would log it as a log.info and then Thor can set an alert if he wishes. Please, check with him on that.